shm_wrapper_aux/shm_wrapper_aux.h

```diff
@@ -0,0 +1,112 @@
+#ifndef SHM_WRAPPER_AUX_H
+#define SHM_WRAPPER_AUX_H
+
+#include <stdio.h>                         //for i/o
+#include <stdlib.h>                        //for standard utility functions (exit, sleep)
+#include <stdint.h>						   //for standard integer types
+#include <sys/time.h>                      //for measuring time
+#include <sys/types.h>                     //for sem_t and other standard system types
+#include <sys/stat.h>                      //for some of the flags that are used (the mode constants)
+#include <fcntl.h>                         //for flags used for opening and closing files (O_* constants)
+#include <unistd.h>                        //for standard symbolic constants
+#include <semaphore.h>                     //for semaphores
+#include <sys/mman.h>                      //for posix shared memory
+#include "../logger/logger.h"              //for logger (TODO: consider removing relative pathname in include)
+#include "../runtime_util/runtime_util.h"  //for runtime constants (TODO: consider removing relative pathname in include)
+
+#define NUM_DESC_FIELDS 6                   //number of fields in the robot description
+#define NUM_GAMEPAD_BUTTONS 17              //number of gamepad buttons
+
+//enumerated names for the buttons on the gamepad
+typedef enum gp_buttons {
+	A_BUTTON, B_BUTTON, X_BUTTON, Y_BUTTON, L_BUMPER, R_BUMPER, L_TRIGGER, R_TRIGGER,
+	BACK_BUTTON, START_BUTTON, L_STICK, R_STICK, UP_DPAD, DOWN_DPAD, LEFT_DPAD, RIGHT_DPAD, XBOX_BUTTON
+} gp_button_t;
+
+//enumerated names for the joystick params of the gamepad
+typedef enum gp_joysticks {
+	X_LEFT_JOYSTICK, Y_LEFT_JOYSTICK, X_RIGHT_JOYSTICK, Y_RIGHT_JOYSTICK
+} gp_joystick_t;
+
+//enumerated names for the different values the robot description fields can take on
+typedef enum robot_desc_vals {
+	ISSUE, NOMINAL,             //values for robot.state
+	IDLE, AUTO, TELEOP,         //values for robot.run_mode
```

Comment:
Could you also add ESTOP?

Reply:
I figured that ESTOP could be implemented by just having the network handler set this field to IDLE, because that's what ESTOP really does. So rather than have you have to deal with doing the same thing for either this thing being in IDLE or ESTOP you just make sure that the student code isn't running when idling.

Reply:
ok thats fine I guess. I was just wondering if there was ever a case where we need to turn the runtime processes off, which would be ESTOP, but hopefully not if our code is robust lol.